MatrixSDK/Crypto/MXCryptoV2.swift

```diff
@@ -101,33 +158,119 @@ public class MXCryptoV2: MXCrypto {
         warnNotImplemented()
     }
     
-    public override func encryptEventContent(_ eventContent: [AnyHashable : Any]!, withType eventType: String!, in room: MXRoom!, success: (([AnyHashable : Any]?, String?) -> Void)!, failure: ((Error?) -> Void)!) -> MXHTTPOperation! {
-        warnNotImplemented()
-        return nil
+    // MARK: - Encrypt / Decrypt
+    
+    public override func encryptEventContent(
+        _ eventContent: [AnyHashable : Any]!,
+        withType eventType: String!,
+        in room: MXRoom!,
+        success: (([AnyHashable : Any]?, String?) -> Void)!,
+        failure: ((Error?) -> Void)!
+    ) -> MXHTTPOperation! {
+        guard let content = eventContent, let eventType = eventType, let roomId = room.roomId else {
+            MXLog.debug("[MXCryptoV2] encryptEventContent: Missing data to encrypt")
+            return nil
+        }
+        
+        guard isRoomEncrypted(roomId) else {
+            MXLog.error("[MXCryptoV2] encryptEventContent: attempting to encrypt event in room without encryption")
+            return nil
+        }
+        
+        MXLog.debug("[MXCryptoV2] encryptEventContent: Encrypting content")
+        
+        Task {
+            do {
+                let users = try await getRoomUserIds(for: room)
```

Comment:
I always wonder if you should fail sending completly if this fails. 
Is it using cache? or always doing a request?